### PR TITLE
Challenge Evaluations Front End

### PR DESCRIPTION
**Pull request by kylekthompson**
This is waiting for #306 to be merged.

@atareshawty let me know how you feel about this so far (the front end stuff in here)

### CODE REVIEW

File: client/src/components/flex/components/flex_child.jsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import styled from 'styled-components';
 
 export default styled.div`
-  flex: ${({ flex }) => flex}
+  ${({ flex }) => flex && `flex: ${flex};`};
```
Comment (atareshawty) [inline client/src/components/flex/components/flex_child.jsx]: When would you want to render a flex child without giving it a flex property?

File: client/src/App.jsx [hunk 2]
```diff
@@ -65,11 +65,11 @@ const App = () =>
       <div id="App-container">
         <Switch>
           <PrivateRoute exact path="/" component={Profile} />
-          {/*<PrivateRoute
-              exact
-              path="/challenges/evaluate"
-              component={ChallengeEvaluations}
-            />*/}
+          <PrivateRoute
```
Comment (atareshawty) [inline client/src/App.jsx]: Yay!!

File: client/src/scenes/challenge/evaluations/index.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+              {currentChallengeId && <Evaluation challenge={currentChallenge} />}
+              {!currentChallengeId && <p>Looks like you don't have any challenges to evaluate!</p>}
+            </FlexChild>
+          </FlexContainer>
+        </FlexChild>
+      </FlexContainer>
+    );
+  }
+
+  setActiveChallengeTo(challengeId) {
```
Comment (atareshawty) [inline client/src/scenes/challenge/evaluations/index.jsx]: This is a slight nit, but my weird preference for component classes is to have the render function last, even after non React specific functions.

File: client/src/scenes/challenge/evaluations/index.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      currentChallengeId = sortedChallenges[0] && sortedChallenges[0].id
+    }
+
+    this.state = {
+      currentChallengeId,
+    };
+  }
+
+  render() {
+    if (!this.props.challenges) {
```
Comment (atareshawty) [inline client/src/scenes/challenge/evaluations/index.jsx]: If you're passing this into the `fetch` util, when would you ever get null?

File: client/src/scenes/challenge/evaluations/index.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      <FlexContainer>
+        <FlexChild>
+          <SideNav>
+            {sortedChallenges.map((challenge) => (
+              <SideNavItem
+                active={currentChallengeId === challenge.id}
+                key={challenge.id}
+                onClick={this.setActiveChallengeTo(challenge.id)}
+                subtitle={challenge.users.map((user) => user.firstName).join(' vs. ')}
+                title={`${challenge.spot.row}${challenge.spot.file}`}
```
Comment (atareshawty) [inline client/src/scenes/challenge/evaluations/index.jsx]: I think there's a spot related util for this. I doubt the logic would change though

File: client/src/scenes/challenge/evaluations/index.jsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+import { FlexChild, FlexContainer } from '../../../components/flex';
+import { helpers } from '../../../data/challenge_evaluations';
+import { helpers as spotHelpers } from '../../../data/spot';
+import { fetch } from '../../../utils';
+import Evaluation from './components/evaluation';
+import SideNav from './components/side-nav';
+import SideNavItem from './components/side-nav-item';
+
+class Evaluations extends React.Component {
+  constructor(props, context) {
```
Comment (atareshawty) [inline client/src/scenes/challenge/evaluations/index.jsx]: You're not actually using context here, right? Just passing it to `super`?

File: client/src/scenes/challenge/evaluations/components/evaluation.jsx [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+const Evaluation = ({ challenge }) => <p>Evaluation {challenge.id}</p>;
```
Comment (atareshawty) [inline client/src/scenes/challenge/evaluations/components/evaluation.jsx]: Could you add some `PropType` validations on this? Wouldn't ESLint yell about this?

File: client/src/components/flex/components/flex_child.jsx (hunk 1 above)
Comment (kylekthompson) [inline client/src/components/flex/components/flex_child.jsx]: For default flex. I can change it back and be explicit with the `flex="0"` though!

File: client/src/scenes/challenge/evaluations/index.jsx (hunk 6 above)
Comment (kylekthompson) [inline client/src/scenes/challenge/evaluations/index.jsx]: Correct! Habit from Typescript-land.

File: client/src/scenes/challenge/evaluations/index.jsx (hunk 4 above)
Comment (kylekthompson) [inline client/src/scenes/challenge/evaluations/index.jsx]: Didn't realize initially that it wasn't rendered until the data was available. I'll get rid of that!

File: client/src/scenes/challenge/evaluations/index.jsx (hunk 3 above)
Comment (kylekthompson) [inline client/src/scenes/challenge/evaluations/index.jsx]: Typescript-land habit again haha. Typically these are private for me, and my linter yells at me if they are before my public render. I'll fix it!

File: client/src/scenes/challenge/evaluations/components/evaluation.jsx (hunk 7 above)
Comment (kylekthompson) [inline client/src/scenes/challenge/evaluations/components/evaluation.jsx]: I was planning on it! Just didn't get around to it yet

**Comment by atareshawty** [comment]
@kylekthompson Thanks for all your work on this. I really appreciate it!!

**Comment by kylekthompson** [comment]
@atareshawty are you good with the prettier changes too? If not I can drop that commit

File: client/src/data/challenge_evaluations/helpers.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    },
+    `Comments were not able to be saved for ${challenge.spot.row}${challenge
+      .spot.file}'s challenge'`
+  );
+const postSavePlaces = challenge =>
+  api.post(
+    '/user_challenges/places',
+    { user_challenges: reduceUserChallengesToIdAndField(challenge, 'place') },
+    `Places were not able to be saved for ${challenge.spot.row}${challenge.spot
+      .file}'s challenge'`
```
Comment (kylekthompson) [inline client/src/data/challenge_evaluations/helpers.js]: Doh 😅

**Comment by atareshawty** [comment]
> are you good with the prettier changes too? If not I can drop that commit

Yup :D Is that the only outstanding issue?

**Comment by kylekthompson** [comment]
> Yup :D Is that the only outstanding issue?

Yep!

**Pull request merged**